sympy/core/relational.py

```diff
@@ -1138,3 +1047,309 @@ def _eval_relation(cls, lhs, rhs):
     '<': StrictLessThan,
     'lt': StrictLessThan,
 }
+
+
+def _n2(a, b):
+    """Return (a - b).evalf(2) if a and b are comparable, else None.
+    This should only be used when a and b are already sympified.
+    """
+    # /!\ it is very important (see issue 8245) not to
+    # use a re-evaluated number in the calculation of dif
+    if a.is_comparable and b.is_comparable:
+        dif = (a - b).evalf(2)
+        if dif.is_comparable:
+            return dif
+
+
+@dispatch(Expr, Expr)
+def _eval_is_ge(lhs, rhs):
+    return None
+
+
+@dispatch(Basic, Basic)
+def _eval_is_eq(lhs, rhs):
+    return None
+
+
+@dispatch(Tuple, Expr)
+def _eval_is_eq(lhs, rhs):  # noqa:F811
+    return False
+
+
+@dispatch(Tuple, AppliedUndef)
+def _eval_is_eq(lhs, rhs):  # noqa:F811
+    return None
+
+
+@dispatch(Tuple, Symbol)
+def _eval_is_eq(lhs, rhs):  # noqa:F811
+    return None
+
+
+@dispatch(Tuple, Tuple)
+def _eval_is_eq(lhs, rhs):  # noqa:F811
+    if len(lhs) != len(rhs):
+        return False
+
+    return fuzzy_and(fuzzy_bool(is_eq(s, o)) for s, o in zip(lhs, rhs))
+
+
+def is_lt(lhs, rhs):
+    """Fuzzy bool for lhs is strictly less than rhs.
+
+    See the docstring for is_ge for more
+    """
+    return fuzzy_not(is_ge(lhs, rhs))
+
+
+def is_gt(lhs, rhs):
+    """Fuzzy bool for lhs is strictly greater than rhs.
+
+    See the docstring for is_ge for more
+    """
+    return fuzzy_not(is_le(lhs, rhs))
+
+
+def is_le(lhs, rhs):
+    """Fuzzy bool for lhs is less than or equal to rhs.
+
+    See the docstring for is_ge for more
+    """
+    return is_ge(rhs, lhs)
+
+
+def is_ge(lhs, rhs):
+    """
+    Fuzzy bool for lhs is greater than or equal to rhs.
+
+    Returns True if lhs is greater than or equal to rhs
+    Returns False if lhs is not greater than rhs
+    Returns None if lhs cannot be compared to rhs, or if the comparison is indeterminate
+
+    is_le calls is_ge swapping rhs and lhs
+    is_lt calls is_ge and then fuzzy nots the result
+    is_gt calls is_le and fuzzy nots the result
```

Comment:
Rather than these details it is better just to say that other which functions are based on this one.